bootstrap/041_quotas.sql

```diff
@@ -0,0 +1,63 @@
+
+create table if not exists internal.quota_task_history(start_time timestamp_ltz, end_time timestamp_ltz, credits_used object, outcome string);
+CREATE OR REPLACE VIEW REPORTING.QUOTA_TASK_HISTORY AS SELECT * FROM INTERNAL.QUOTA_TASK_HISTORY;
+
+-- returns SQL which reads the query_history table function and aggregates the credits used by both user and role. executed within a task
+-- as an owner's rights procedure cannot call the query_history table function.
+create or replace procedure internal.get_daily_quota_select()
+returns string
+language sql
+as
+BEGIN
+    let s string := $$
+-- TODO this recomputes the usage since midnight every time it runs. We should cache this for the day and update quotas based on deltas.
+-- TODO the table func caps out at 10k rows which could miss queries for busy accounts.
+with todays_queries as(
+    select
+        total_elapsed_time,
+        credits_used_cloud_services,
+        warehouse_size,
+        user_name,
+        role_name
+    from table(information_schema.query_history(RESULT_LIMIT => 10000, END_TIME_RANGE_START => date_trunc('day', current_timestamp()), END_TIME_RANGE_END => current_timestamp()))
+    where end_time > date_trunc('day', current_timestamp())
```

Comment:
At some point I convinced myself this was necessary as I was seeing incomplete queries (which were being returned with an `END_DATE` of the unix epoch). However, this might have been me testing without the `END_TIME_RANGE_*` attributes. Let me dbl check and either comment or remove.